app/components/Form/TimePicker.js

```diff
@@ -27,57 +27,98 @@ type Props = {
 };
 
 type State = {
-  value: moment
+  value: moment,
+  tempValue: {
+    hour: string,
+    minute: string
+  }
 };
 
 function parseValue(value) {
   return value ? moment(value) : moment();
 }
 
 class TimePicker extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    const value = parseValue(this.props.value);
+    this.state = {
+      value,
+      tempValue: {
+        hour: value.format('HH'),
+        minute: value.format('mm')
+      }
+    };
+  }
+
   static defaultProps = {
-    value: ''
+    value: '',
+    tempValue: {
+      hour: '',
+      minutes: ''
+    }
   };
 
   static Field: any;
 
-  state = {
-    value: parseValue(this.props.value)
-  };
-
-  onNext = (unit: string) => () => {
-    this.setState(
-      prevState => ({
-        value: prevState.value.clone().add(1, unit)
-      }),
-      this.commit
-    );
+  onNext = (unit: string) => (e: SyntheticInputEvent<HTMLInputElement>) => {
+    this.setState(prevState => {
+      const value = prevState.value.clone().add(1, unit);
+      return {
+        value,
+        tempValue: {
+          hour: value.format('HH'),
+          minute: value.format('mm')
+        }
+      };
+    }, this.commit);
   };
 
   onNextHour = this.onNext('hour');
   onNextMinute = this.onNext('minute');
 
   onPrev = (unit: string) => () => {
-    this.setState(
-      prevState => ({
-        value: prevState.value.clone().subtract(1, unit)
-      }),
-      this.commit
-    );
+    this.setState(prevState => {
+      const value = prevState.value.clone().subtract(1, unit);
+      return {
+        value,
+        tempValue: {
+          hour: value.format('HH'),
+          minute: value.format('mm')
+        }
+      };
+    }, this.commit);
   };
 
   onPrevHour = this.onPrev('hour');
   onPrevMinute = this.onPrev('minute');
 
-  onChange = (unit: 'hour' | 'minute') => () => {
-    this.setState(
-      {
-        value: this.state.value.clone()[unit].call(null, 1)
-      },
-      this.commit
-    );
+  onChange = (unit: 'hour' | 'minute') => (
+    e: SyntheticInputEvent<HTMLInputElement>
+  ) => {
+    const { value } = e.target;
+    if (Number(value) || !Number(value)) {
+      if (
+        (unit === 'hour' && Number(value) < 24) ||
+        (unit === 'minute' && Number(value) < 60)
+      ) {
```

Comment:
What happens if you write something like `-1`?

Reply:
You cannot write -1 specifically, since only numbers are allowed in the field. However, by copy pasting `-1` into the field will cause some weird behavior. I think this goes under the category "user error" and should not be handled. :):)